Algorithms/Sequences/EulerTotientSequence.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+namespace Algorithms.Sequences
+{
+    /// <summary>
+    ///     <para>
+    ///         Sequence of Euler totient function phi(n).
+    ///     </para>
+    ///     <para>
+    ///         Wikipedia: https://en.wikipedia.org/wiki/Euler%27s_totient_function.
+    ///     </para>
+    ///     <para>
+    ///         OEIS: https://oeis.org/A000010.
+    ///     </para>
+    /// </summary>
+    public class EulerTotientSequence : ISequence
+    {
+        /// <summary>
+        ///     <para>
+        ///         Gets sequence of Euler totient function phi(n).
+        ///     </para>
+        /// </summary>
+        public IEnumerable<BigInteger> Sequence
+        {
+            get
+            {
+                yield return BigInteger.One;
+
+                for (BigInteger i = 2; ; i++)
+                {
+                    var result = i;
+                    var n = i;
+
+                    var factors = PrimeFactors(i);
+                    foreach (var factor in factors)
+                    {
+                        while (n % factor == 0)
+                        {
+                            n /= factor;
+                        }
+
+                        result -= result / factor;
+                    }
+
+                    if (n > 1)
+                    {
+                        result -= result / n;
+                    }
```

Comment:
I'm a bit lost as to what's happening here, could you add some comments or a reference to which formula is used? And why not use Euler's product formula? It looks simpler IMO.
![image](https://user-images.githubusercontent.com/29201949/136148606-12f93699-3536-45c2-9e2c-227327026d89.png)

Reply:
Expanded in comments